### PR TITLE
Preprocess gradientTransform translation

### LINKED ISSUE #127: Apply gradient translation
Noto has things like:

```xml
<linearGradient id="c" x1="63.85" x2="63.85" y1="4245" y2="4137.3" gradientTransform="translate(0 -4122)" gradientUnits="userSpaceOnUse">
<radialGradient id="b" cx="63.523" cy="12368" r="53.477" gradientTransform="matrix(1 0 0 .2631 0 -3150)" gradientUnits="userSpaceOnUse">
```

At minimum, just apply the translate transform on a gradient. Perhaps just apply the whole transform, need to think about that.

Credit to @anthrotype for spotting this pattern.

### PR DESCRIPTION
**Pull request by rsheeter**
Early steps toward surviving some of the weirdness in Noto gradients (weird gradient coords + transforms) better.

Move parsing logic nanoemoji => picosvg to enable.

Fixes #127

### CODE REVIEW

File: src/picosvg/svg.py [hunk 1]
```diff
@@ -679,17 +702,89 @@ def remove_attributes(self, names, xpath="/svg:svg", inplace=False):
...
+                assert r1 == a * x + c * y + dx
+                assert r2 == b * x + d * y + dy
+                y_prime = (r2 - r1 * b / a) / (d - b * c / a)
+
+                # Sub y` into 1)
+                # 1) x` = (r1 - cy`) / a
+                x_prime = (r1 - c * y_prime) / a
+
+                # sanity check: a`(x`, y`) should be a(x, y)
+                p = affine.map_point((x, y))
```
Comment (anthrotype) [inline src/picosvg/svg.py:768]: nit: `p` is the same as `(r1, r2)` above, no need to recompute here

File: tests/svg_test.py [hunk 2]
```diff
@@ -403,3 +403,47 @@ def test_apply_style_attributes(actual, expected_result):
...
+            '<radialGradient id="b" cx="63.523" cy="12368" r="53.477" gradientTransform="matrix(.5 0 0 .2631 0 -3150)" gradientUnits="userSpaceOnUse"/>',
+            '<radialGradient id="b" cx="63.523" cy="395.366021" r="53.477" gradientTransform="matrix(0.5 0 0 0.2631 0 0)" gradientUnits="userSpaceOnUse"/>',
+        ),
+        # Real example from emoji_u1f44d.svg
+        # Using all 6 parts
+        (
+            '<radialGradient id="d" cx="2459.4" cy="-319.18" r="20.331" gradientTransform="matrix(-1.3883 .0794 -.0374 -.6794 3505.4 -353.39)" gradientUnits="userSpaceOnUse"/>',
+            '<radialGradient id="d" cx="-71.60264" cy="-94.82264" r="20.331" gradientTransform="matrix(-1.3883 0.0794 -0.0374 -0.6794 0 0)" gradientUnits="userSpaceOnUse"/>',
+        ),
+    ],
```
Comment (anthrotype) [inline tests/svg_test.py:434]: should you maybe also add a test with gradientUnits="objectBoundingBox" even if relatively less frequent than "userSpaceOnUse"?

File: src/picosvg/svg.py [hunk 3]
```diff
@@ -46,6 +63,10 @@
 _DEFAULT_DEFAULT_TOLERENCE = 0.1
 
 
+# Rounding for rewritten gradient matrices
+_GRADIENT_TRANSFORM_NDIGITS = 6
```
Comment (anthrotype) [inline src/picosvg/svg.py:67]: :+1 I think 6 decimal digits is more than enough

**Review by anthrotype — APPROVED** [review]
Thanks for working on this! i think the math checks out.

File: src/picosvg/svg.py (hunk 1 above)
Comment (rsheeter) [reply src/picosvg/svg.py:768]: Ty, fixing

File: tests/svg_test.py (hunk 2 above)
Comment (rsheeter) [reply tests/svg_test.py:434]: Done

**Pull request merged**